### PR TITLE
Fix race conditions in updating RemoteTaskStats

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description

In the existing implementation, the shared currentRequestStartNanos object
may be updated by the next request after the current request future is done
but before the callback handler has read the value of currentRequestStartNanos
set before the current request was started.

## General information

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Core query engine

> How would you describe this change to a non-technical end user or system administrator?

Fixes potential inaccuracies in recording of RemoteTaskStats

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

**Comment by findepi** [comment]
@raunaqmorarka is this related to https://github.com/trinodb/trino/issues/11039 ?

**Comment by raunaqmorarka** [comment]
@findepi I don't think this will resolve the `value is negative` failures. It's just going to avoid inaccuracies in stats due to a possible race condition.

File: core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java [hunk 1]
```diff
@@ -158,52 +155,62 @@ private synchronized void fetchDynamicFiltersIfNecessary()
...
-        try (SetThreadName ignored = new SetThreadName("DynamicFiltersFetcher-%s", taskId)) {
-            updateStats(currentRequestStartNanos.get());
-            try {
-                updateDynamicFilterDomains(newDynamicFilterDomains);
-                errorTracker.requestSucceeded();
-            }
-            finally {
-                fetchDynamicFiltersIfNecessary();
-            }
+        private final long currentRequestStartNanos;
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java:164]: same

File: core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java [hunk 2]
```diff
@@ -247,49 +241,56 @@ synchronized void updateTaskInfo(TaskInfo newTaskInfo)
...
     }
 
-    @Override
-    public void success(TaskInfo newValue)
+    private class TaskInfoResponseCallback
+            implements SimpleHttpResponseCallback<TaskInfo>
     {
-        try (SetThreadName ignored = new SetThreadName("TaskInfoFetcher-%s", taskId)) {
-            lastUpdateNanos.set(System.nanoTime());
+        private final long currentRequestStartNanos;
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java:247]: same

**Review by losipiuk — APPROVED** [review]

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
Merged, thanks!

**Pull request merged**